src/cart/crt_iv.c

```diff
@@ -2879,9 +2872,9 @@ crt_iv_update_internal(crt_iv_namespace_t ivns, uint32_t class_id,
 put:
 	iv_ops->ivo_on_put(ivns, NULL, priv);
 exit:
-	if (rc != DER_SUCCESS) {
+	if (ivns_internal != NULL) {
```

Comment:
this is not quite right. its possible that we hit line 2860 on error condition in which case ivns will have +2 ref count (once from initial lookup, once from newly added line 2848, but only once decref is done here in this case

Reply:
Indeed. An IVNS_DECREF(cb_info->uci_ivns_internal) should be added around line 2858.